### ISSUE TITLE
MacOS Install Error

### ISSUE THREAD

**Issue by diogoamore** [GeoscienceAustralia/anuga_core#198]
Dear admin,

I ve been trying to install Anuga on my MacOS but have not had much success. I have installed on my Ubuntu alright. However, as I have tried the install_conda_macos.sh file, I had no success. I have used python2 command instead of python throughout the command list and the installation seems to go all the way to the end. But as I try to runtests.py, lots of errors pop up. I haven't gone through all of them because I am not sure whether there are incompatibilities related to python scripts and Mac. 
If there are any specific detail I should care about while attempting an install on MacOS I would much appreciate, as I have not read anything at Github description page. 

Cheers
Diogo

**Comment by stoiver** [comment]
Dear @diogoamore. We have continuous integration testing for the MacOS using travis CI, which seems to be working. 

Having a look at the script install_conda_macos.sh which is used on travis.ci I can't see where the compiler is set. 

Could you check that in your installation that the gcc compiler is used. If not then that might be your problem. 

Looking at the page https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html it seems that you/we might need to install a compiler package from conda. 

it would be great if you could let us know how that works so that we can add a page in our wiki
https://github.com/GeoscienceAustralia/anuga_core/wiki

**Comment by diogoamore** [comment]
Dear @stoiver,

I have managed to finally install anuga_core after several attempts to run the shell commands individually, but not with parallel working yet though. Not sure what the issue is yet with parallel. Nevertheless, it seems to be working already for the simple_examples folder tests.
 
The issue now is with anuga_viewer. I ve spent quite some time on it but there are several issues with c++ version. I tried to add the CFLAG=std=c++11 line within test, swwreader and viewer MAKEFILES, however, it did not do the trick yet, as folder issues showed up (mostly with these folder: -L/usr/X11R6/lib -L/sw/lib -L/usr/local/lib64). 

As I am not a programmer myself, only a very motivated functional programmer I would say, I think I reached my limit here. If there is anything you can tell about both parallel install issue on Mac and the further anuga_viewer install issues, I would very much appreciate the support.

Best
Diogo

**Comment by stoiver** [comment]
@diogoamore  we should probably break this into two issues, one on the anuga_core installation on MacOS and one on the installation of the anuga-viewer on MacOS. 

Regarding the installation of anuga_core on MacOS. Can you provide the steps you took to get it installed. From that we can create an installation page on our wiki. 

Perhaps @uniomni can provide some help with the installation of the parallel version. 

By the way you can also use the separate mail list  anuga-user@lists.sourceforge.net   (subscription via ​https://lists.sourceforge.net/lists/listinfo/anuga-user)

**Comment by diogoamore** [comment]
Dear @stoiver,

Apologies for the late reply. I was busy with some other stuff in the past few days. We can definitely break the issue into two as you suggested.

Regarding the correct steps I took to effectively install anugga_core on my MacOS, I basically went through the install_conda_macos.sh file command list and ran them one by one from scratch after the first failed attempt to install anuga_core. So, I believe that the final installation success was due to making sure both `pip` and `python` commands were both associated with the correct python version needed by Anuga, and making sure I did not install Miniconda since I already had Anaconda installed, and the attempt to install miniconda was not an efficient step within the bash file. So perhaps removing this Miniconda install step from bash and leaving a warning msg somewhere along code the lines or within github install page could do the job.

I also believe the `conda` command along with `pip` and `python` also produced some conflict during my first attempt due to different python versions installed and which python version each command was associated with. So, even if I `PYTHON_VERSION=${PYTHON_VERSION:-"2.7"}` was set within the bash file, I think my `pip` command was associated with python3.X, hence, this command (`conda create -n anuga_env -c conda-forge --yes python=$PYTHON_VERSION pip numpy scipy netcdf4 nose matplotlib gdal dill`) did not install the packages correctly. So, when I started all from scratch paying close attention to every step, and using system `pip` and `python` commands rather than `conda` commands, things seemed to have gone the right way. But it was me, some other people might not have any issues with Anaconda. I admit I am not really used to Anaconda yet, so that might the problem.

Lastly, I believe your bash script is very well set, however, a couple of notes pointing to the fact that if one has Anaconda already installed, or if the bash script run through some errors, it would be best running the commands step-by-step so that one is clear that the install process is running smoothly. 

Ubuntu bash script did not present errors I believe because it does not use anaconda, and `apt` command did the job smoothly. I have recently had problems with anaconda installing other software. So, I think a warning regarding usage of Anaconda or Miniconda with the correct system python versions might help someone who might have different python and anaconda versions installed that could produce version conflict.

Hope this clarify my issue. I wil try to open an issue regarding anuga_viewer in a mailing list.

Best,
Diogo
